### PR TITLE
Add CLI instructions for LSO

### PR DESCRIPTION
**Pull request by bobfuru**
[BZ 1800649](https://bugzilla.redhat.com/show_bug.cgi?id=1800649)
For 4.2+

### CODE REVIEW

**Comment by bobfuru** [comment]
This PR adds CLI instructions for Local Storage Operator. Preview build link is here: https://bz1800649--ocpdocs.netlify.com/openshift-enterprise/latest/storage/persistent_storage/persistent-storage-local.html#local-storage-install_persistent-storage-local

@gnufied PTAL - this came from a Dev request you received from @achuzhoy. 

@liangxia PTAL for QE

File: modules/persistent-storage-local-install.adoc [hunk 1]
```diff
@@ -49,3 +49,74 @@ $ oc annotate project local-storage openshift.io/node-selector=''
...
++
+----
+$ oc apply -f local-storage.yaml
+----
++
+At this point, the Operator Lifecycle Manager (OLM) is now aware of the Local Storage Operator. A ClusterServiceVersion (CSV) for the Operator should appear in the target namespace, and APIs provided by the Operator should be available for creation.
++
+. Verify local storage installation by checking that all Pods and the Local Storage Operator have been created:
+
+.. Check that all the required Pods have been created:
```
Comment (bobfuru) [inline modules/persistent-storage-local-install.adoc:106]: Not sure if this step (`oc get pods`) is really necessary for verifying that LSO is working?

**Comment by bobfuru** [comment]
@gnufied You are correct, the YAML needs local-operator-group. I've updated to use your file. Tested, and it works. PTAL and let me know if this looks good to you. Thanks.
[Preview doc link](https://bz1800649--ocpdocs.netlify.com/openshift-enterprise/latest/storage/persistent_storage/persistent-storage-local.html#local-storage-install_persistent-storage-local)

File: modules/persistent-storage-local-install.adoc [hunk 2]
```diff
@@ -49,3 +49,85 @@ $ oc annotate project local-storage openshift.io/node-selector=''
...
++
+.Example local-storage
+[source,yaml]
+----
+  apiVersion: v1
+  kind: Namespace
+  metadata:
+    name: local-storage
+  ---
+  apiVersion: operators.coreos.com/v1alpha2
```
Comment (bobfuru) [inline modules/persistent-storage-local-install.adoc:81]: @gnufied Is `v1alpha2` correct here?

**Comment by liangxia** [comment]
@qinpingli  PTAL

File: modules/persistent-storage-local-install.adoc (hunk 2 above)
Comment (gnufied) [reply modules/persistent-storage-local-install.adoc:81]: yeah seems correct.

File: modules/persistent-storage-local-install.adoc [hunk 3]
```diff
@@ -49,3 +49,85 @@ $ oc annotate project local-storage openshift.io/node-selector=''
...
+    channel: "4.2"
+    installPlanApproval: Automatic
+    name: local-storage-operator <1>
+    source: redhat-operators <2>
+    sourceNamespace: openshift-marketplace <3>
+----
+<1> Name of the Operator to subscribe to.
+<2> Name of the CatalogSource that provides the Operator.
+<3> Namespace of the CatalogSource. Use `openshift-marketplace` for the default
+OperatorHub CatalogSources.
```
Comment (gnufied) [inline modules/persistent-storage-local-install.adoc:105]: Essentially the only field that user should edit here is to change the channel to match OCP version. Everything else user *must* not touch.

Comment (qinpingli) [reply modules/persistent-storage-local-install.adoc:105]: In here, the end users should care about the spec.channel and spec.installPlanApproval same with ui installation.

File: modules/persistent-storage-local-install.adoc [hunk 4]
```diff
@@ -49,3 +49,85 @@ $ oc annotate project local-storage openshift.io/node-selector=''
...
+
+.. Check the ClusterServiceVersion (CSV) YAML manifest to see that the Local Storage Operator is available in the `local-storage` project:
++
+----
+$ oc get csvs -n local-storage
+NAME                                         DISPLAY         VERSION               REPLACES   PHASE
+local-storage-operator.4.2.26-202003230335   Local Storage   4.2.26-202003230335              Succeeded
+----
+
+After all Pods are ready and you have verified that the Local Storage Operator is available, you can begin using local storage to store data from your cluster.
```
Comment (qinpingli) [inline modules/persistent-storage-local-install.adoc:133]: After the local storage operator is deployed, we still need to create `localvolume` object to install the local storage provisioner to help provide PVs.

File: modules/persistent-storage-local-install.adoc (hunk 1 above)
Comment (qinpingli) [reply modules/persistent-storage-local-install.adoc:117]: keep this is ok for me.

File: modules/persistent-storage-local-install.adoc (hunk 4 above)
Comment (bobfuru) [reply modules/persistent-storage-local-install.adoc:133]: > After the local storage operator is deployed, we still need to create `localvolume` object to install the local storage provisioner to help provide PVs.

This procedure (creating local volume resource) is described in the next section immediately following Operator install: 
https://docs.openshift.com/container-platform/4.4/storage/persistent_storage/persistent-storage-local.html#local-volume-cr_persistent-storage-local

File: modules/persistent-storage-local-install.adoc (hunk 3 above)
Comment (bobfuru) [reply modules/persistent-storage-local-install.adoc:105]: Fair point, @gnufied. I was pulling the annotation descriptions from our Operator docs ([link](https://bz1800649--ocpdocs.netlify.com/openshift-enterprise/latest/operators/olm-adding-operators-to-cluster.html#olm-installing-operator-from-operatorhub-using-cli_olm-adding-operators-to-a-cluster)).
But we don't want to give any impression that those fields are customizable, so I removed them. I also added a description for channel (to match OCP version).

@qinpingli I didn't add a note to spec.installPlanApproval as I wasn't sure we wanted to have them use anything but `Automatic`. Please suggest wording for this field if you feel strongly that a description is needed.

**Comment by bobfuru** [comment]
@gnufied and @qinpingli - I've made revisions based on your comments. Please add `lgtm` if you are OK with changes. Thanks.

File: modules/persistent-storage-local-install.adoc (hunk 4 above)
Comment (qinpingli) [reply modules/persistent-storage-local-install.adoc:130]: @bobfuru got it.  `you can begin using local storage to store data from your cluster` misleads me, how about change to something like `After all the check passed, local storage operator is installed successfully.`

Comment (bobfuru) [reply modules/persistent-storage-local-install.adoc:130]: @qinpingli Sure - that makes sense. I changed to your suggestion.

File: modules/persistent-storage-local-install.adoc [hunk 5]
```diff
@@ -49,3 +49,82 @@ $ oc annotate project local-storage openshift.io/node-selector=''
...
+    targetNamespaces:
+      - local-storage
+  ---
+  apiVersion: operators.coreos.com/v1alpha1
+  kind: Subscription
+  metadata:
+    name: local-storage-operator
+    namespace: local-storage
+  spec:
+    channel: "stable-{product-version}" <1>
```
Comment (bobfuru) [inline modules/persistent-storage-local-install.adoc:96]: @qinpingli I found this in our docs: https://docs.openshift.com/container-platform/4.2/updating/updating-cluster-between-minor.html#understanding-upgrade-channels_updating-cluster-between-minor

So I added `stable-` in channel version. OK?

**Comment by bobfuru** [comment]
@qinpingli One question inline for you about `spec.channel`. If that and my update to last sentence in the file are okay with you, can I please get a LGTM?

File: modules/persistent-storage-local-install.adoc (hunk 5 above)
Comment (qinpingli) [reply modules/persistent-storage-local-install.adoc:96]: @bobfuru These channels should be for payload image, for 3rd level operators, the channels are `4.2`, '4.3', ...

Comment (bobfuru) [reply modules/persistent-storage-local-install.adoc]: Ah, I see @qinpingli - thanks for clarifying. Changed it back to just `{product-version}` (dropped `stable-`). Does everything look good to you now?

**Comment by qinpingli** [comment]
@bobfuru  Thanks for the update.
LGTM now.

**Comment by bobfuru** [comment]
@gnufied PTAL to quickly confirm that this looks good to you with revisions?

**Comment by bobfuru** [comment]
@openshift/team-documentation PTAL

**Comment by bmcelvee** [comment]
@bobfuru sorry for the delay on this - LGTM!

**Pull request merged**